Detectors/TRD/simulation/test/testPileupTool.cxx

```diff
@@ -0,0 +1,108 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#define BOOST_TEST_MODULE Test TRD Pileup Tool
+#define BOOST_TEST_MAIN
+#define BOOST_TEST_DYN_LINK
+#include <boost/test/unit_test.hpp>
+
+#include "DataFormatsTRD/Constants.h"
+#include "DataFormatsTRD/SignalArray.h" // for SignalContainer
+#include "TRDSimulation/PileupTool.h"
+
+namespace o2
+{
+namespace trd
+{
+
+BOOST_AUTO_TEST_CASE(TRDPileupTool_test)
+{
+  PileupTool tool;
+
+  auto fill_ones = [](std::array<float, constants::TIMEBINS>& signals) {
+    for (int tb = 0; tb < constants::TIMEBINS; ++tb) {
+      signals[tb] = 1;
+    }
+  };
+
+  double triggerTime;
+  std::vector<SignalArray> signalArrays(3);
+  std::array<SignalContainer, constants::MAXCHAMBER> chamberSignals;
+  std::deque<std::array<SignalContainer, constants::MAXCHAMBER>> pileupSignals;
+
+  // set the signals
+  signalArrays[0].firstTBtime = 0;
+  signalArrays[1].firstTBtime = 2000;
+  signalArrays[2].firstTBtime = 4900;
+  fill_ones(signalArrays[0].signals);
```

Comment:
Fyi: You could just use `std::fill(signalArrays[0].signals.begin(), signalArrays[0].signals.end(), 1)`

Reply:
Thanks! I forgot about `std::fill`. I will use it instead of that over though lambda.